src/Base/Interval.hs

```diff
@@ -23,11 +29,17 @@ import Base.PitchClass
 
 import Common.Utils (modByFrom)
 
-import Data.Maybe (fromJust)
+import Data.Maybe (isJust, fromJust)
 
 data Interval = Interval { getQuality :: Quality
                          , getSize :: Int
-                         }
+                         } deriving (Generic)
+
+instance Arbitrary Interval where
+  arbitrary = suchThat (genericArbitrary uniform) (isJust . intervalToDistance)
+
+propInvertSum :: Interval -> Bool
+propInvertSum i = (invert i) |+| i == (intervalFrom Perfect 1)
```

Comment:
I believe all of the parentheses here are redundant.

Reply:
Fixed